pkg/util/metrics_helper_test.go

```diff
@@ -95,6 +95,45 @@ func TestGetMetricsWithLabelNames(t *testing.T) {
 	}, out2)
 }
 
+func BenchmarkGetMetricsWithLabelNames(b *testing.B) {
+	const (
+		numMetrics         = 1000
+		numLabelsPerMetric = 10
+	)
```

Comment:
**thought:** I'd be interested to see how the benchmark results change if you increase these values a bit. I'd imagine the effect is amplified for higher cardinality queries

Reply:
> higher cardinality queries

This helper is not used by queries but to aggregate metrics exposed by ` /metrics`. 1K for a single aggregation is below the number in the cluster I'm profiling, but above the average in most Cortex installations.